ROADMAP.md

```diff
@@ -2,23 +2,100 @@
 
 This document is meant to provide transparency in planning and implementation goals for the Koop project.
 
-## July 2015
-
-### Changes 
-* FeatureService optimizations 
-  * Make smarter requests to the DB based on queries 
-  * clean up the logic to be easier to grok
-* Refactor the lib/* files on a whole
-  - We want pull out as much non-koop specific code as possible into separate modules
-* Add support for the simple style spec - https://github.com/mapbox/simplestyle-spec
-* Address consistency issues in method calls to the Cache from providers 
-  - be clear and consitent about passing table names instead of "types" and "keys"
-
-## September 2015
-
-### Changes 
-* Explore removing ogr2ogr as a dep for exporting file formats
-### Additions 
-* Support for exporting FGDB
-* Support for OGC WMS and WFS 
-* A koop desktop app built with something like Electron
+## v3
+
+Major API changes:
+
+Provide better support for `plugins`. Right now the model for creating a plugin is pretty poor, as illustrated by https://github.com/koopjs/koop/issues/195.
+
+Move away from `provider` in favor of three new types of plugins: `source`, `export`, and `interface`.
+
+This can be thought of more or less like so:
+
+`server/koop/:source{/**}/:item{.:export,/:interface}`
+
+### Source
+
+Defines methods for extracting data from a third party source and transforming it into standard GeoJSON. Handles registering hosts if necessary.
+
+This definition is looser than a provider in that it encompasses any kind of data source, not just open data providers.
+
+Ideally a `source` plugin is more focused, meaning that
+
+- (b) it is only concerned with the specifics of its data source and how to hand that data over to koop
+- (c) it does not interact directly with the koop cache in any way
+- (a) it is less tightly coupled to express and has as little knowledge of server requests and responses as necessary
+
+Addresses:
+
+* `lib/` refactor
+* better encapsulation
+* BaseModel & BaseController issues & limitations
+
+Module naming convention:
+
+```
+koop-source-*
+
+e.g.
+
+koop-source-agol
+koop-source-socrata
+koop-source-ckan
+```
+
+Methods:
+
+* extract (internal method, can be used by itself, e.g. `require(koop-source-agol/extract)`)
+* transform (internal method, can be used by itself, e.g. `require(koop-source-agol/transform)`)
+* fetch (primary method used by koop core to request data from source)
+
+### Export
+
```

Comment:
Agreed. Maybe this can be a more short-term focus as it does seem to be a necessity. This could also give us independent modules for transforming GeoJSON into a Shapefile, CSV, KML, FGDB etc (note that some of these already exist) that would be useful outside of the project.